module/zfs/spa.c

```diff
@@ -4274,6 +4459,10 @@ spa_tryimport(nvlist_t *tryconfig)
 				kmem_free(dsname, MAXPATHLEN);
 			}
 			kmem_free(tmpname, MAXPATHLEN);
+
+			/*
+			 * Should we now clear error if it is set to EEXIST?
+			 */
```

Comment:
Yes, but it looks like you can just unconditionally set `error = 0` in this block.